src/DIPS.Xamarin.UI/Controls/Popup/PopupBehaviour.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Windows.Input;
+using DIPS.Xamarin.UI.Extensions;
+using Xamarin.Forms;
+
+namespace DIPS.Xamarin.UI.Controls.Popup
+{
+    public class PopupBehaviour : Behavior<View>
+    {
+        private readonly Command m_onTappedCommand;
+        private View? m_attachedTo;
+        public PopupBehaviour()
+        {
+            m_onTappedCommand = new Command(ShowPopup);
+        }
+
+        protected override void OnAttachedTo(View bindable)
+        {
+            m_attachedTo = bindable;
+            if (bindable is Button button)
+            {
+                button.Command = m_onTappedCommand;
+            }
+            else
+            {
+                bindable.GestureRecognizers.Add(new TapGestureRecognizer { Command = m_onTappedCommand });
+            }
+
+            base.OnAttachedTo(bindable);
+        }
+
+        protected override void OnDetachingFrom(View bindable)
+        {
+            base.OnDetachingFrom(bindable);
+        }
+
+        public static readonly BindableProperty CloseOnClickProperty =
+            BindableProperty.CreateAttached("CloseOnClick", typeof(bool), typeof(PopupBehaviour), false, propertyChanged: OnCloseOnClickChanged);
+
+        public static void OnCloseOnClickChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            var view = (View)bindable;
+            view.BindingContextChanged += (s, e) =>
+            {
+                var popupLayout = view.GetParentOfType<PopupLayout>();
+                if (popupLayout != null)
+                {
+                    popupLayout.AddOnCloseRecognizer(view);
+                }
+            };
+        }
+
+        public static void SetCloseOnClick(BindableObject view, bool value)
+        {
+            view.SetValue(CloseOnClickProperty, value);
+        }
+
+        public static bool GetCloseOnClick(BindableObject view)
+        {
+            return (bool)view.GetValue(CloseOnClickProperty);
+        }
+
+        public static readonly BindableProperty PopupBindingContextFactoryProperty =
```

Comment:
Maybe this can be named `BindingContextFactory` /`BindingContextFunc` / `BindingContext`?